backend/oma/src/test/java/com/oma/services/AddressServiceTest.java

```diff
@@ -0,0 +1,127 @@
+    }
+
+    @Test
+    public void shouldGetAllAddressesFromDB(){
+//        given
+        Address[] expected = { new Address("address1"), new Address("address2"), new Address("address3")};
+//        when
+        for(Address address : expected){
+            session.beginTransaction();
+            session.save(address);
+            session.getTransaction().commit();
+        }
+        Address[]result = new Address[expected.length];
+        result = addressService.getAllAddresses()
+                .stream()
+                .sorted(Comparator.comparingLong(Address::getId))
+                .collect(Collectors.toList())
+                .toArray(result);
+
+//        then
+        assertArrayEquals(expected, result, "AddressService get all addresses from DB");
+    }
+
+    @Test
+    public void shouldGetEmptyListWithAddressesFromDB(){
+//        when
+        List<Address> result = addressService.getAllAddresses();
+//        then
+        assertEquals(result, Collections.EMPTY_LIST);
+    }
+
+    @Test
+    public void shouldSaveAddressInDB(){
+//        given
+        Address address = new Address("streetExample","zipCodeExample", "cityEample", 900900900);
+//        when
+        addressService.saveAddress(address);
+        List<Address> addresses = addressService.getAllAddresses();
+//        then
+        assertTrue(addresses.contains(address));
+    }
+
+    @Test
+    public void shouldGetAddressFromDBForId(){
+//        given
+        Address address = new Address("example","zipcode","city", 900900900);
+//        when
+        addressService.saveAddress(address);
+        long id = address.getId();
+        Address result = addressService.getAddressForId(id);
+//        then
+        assertEquals(address, result);
+    }
+
+    @Test
+    public void shouldCatchExceptionForIncorrectIdNumber(){
+//        given
+        long id= addressService.getAllAddresses().stream().mapToLong(Address::getId).sum();
+//        then
+        assertThrows(Exception.class, ()->addressService.getAddressForId(id));
+
+    }
+
+    @Test
+    public void shouldUpdateAddressWithId(){
+//        given
+        Address input = new Address("Basic Input");
+        Address update = new Address("Updated Input");
+//        when
+        addressService.saveAddress(input);
+        addressService.updateAddressForId(input.getId(), update);
+        Address expected = addressService.getAddressForId(input.getId());
+//        then
+        assertEquals(expected.getStreetNameAndNumber(), update.getStreetNameAndNumber());
+    }
+    @Test
+    public void shouldRemoveAddressWithIdFromDB(){
+//        given
+        Address addressToRemove = new Address("AddressToRemove");
+//        when
+        addressService.saveAddress(addressToRemove);
+        long id = addressToRemove.getId();
+        addressService.removeAddressWithId(id);
+//        then
+        assertThrows(Exception.class, ()-> addressService.getAddressForId(id));
+    }
```

Comment:
Maybe extra test with wrong email, without necessary symbol or domain?

Reply:
I know what u mean but this is service for Address , like company address :) . I will write some test for wrong email address in email engine :+1: